kolibri_explore_plugin/collectionviews.py

```diff
@@ -421,8 +415,33 @@ def get_status(self):
                     + progress_per_channel * current_job_factor
                 )
 
+        elif self._stage == DownloadStage.APPLYING_EXTERNAL_TAGS:
+            if total_tasks_number > 0:
+                progress = (
+                    PROGRESS_STEPS["downloading"]
+                    + (
+                        PROGRESS_STEPS["tagging"]
+                        - PROGRESS_STEPS["downloading"]
+                    )
+                    * current_task_number
+                    / total_tasks_number
+                )
+            else:
+                progress = 0
```

Comment:
I would say "else, progress is the previous step progress" instead of zero. Trying to prevent the progress bar to go back. Note, this was wrong before, you are just carrying the zero from the previous code.
```suggestion
                progress = PROGRESS_STEPS["downloading"]
```